Libraries/pbxbuild/Sources/Phase/ProductTypeResolver.cpp

```diff
@@ -268,6 +269,135 @@ ResolveFrameworkStructure(Phase::Environment const &phaseEnvironment, Phase::Con
     return true;
 }
 
+static ext::optional<std::string>
+FindUmbrellaHeaderName(pbxsetting::Environment const &environment, std::string const &moduleName, pbxproj::PBX::Target::shared_ptr const &target)
+{
+    std::string lowerModuleName;
+    std::transform(moduleName.begin(), moduleName.end(), std::back_inserter(lowerModuleName), ::tolower);
+
+    ext::optional<std::string> umbrellaHeader;
+    for (pbxproj::PBX::BuildPhase::shared_ptr const &buildPhase : target->buildPhases()) {
+        if (buildPhase->type() != pbxproj::PBX::BuildPhase::Type::Headers) {
+            continue;
+        }
+
+        for (pbxproj::PBX::BuildFile::shared_ptr const &buildFile : buildPhase->files()) {
+            if (buildFile->fileRef() == nullptr) {
+                continue;
+            }
+
+            std::string filePath = environment.expand(buildFile->fileRef()->resolve());
+            if (FSUtil::GetFileExtension(filePath) == "h") {
+                std::string baseName = FSUtil::GetBaseNameWithoutExtension(filePath);
+
+                std::string lowerBaseName;
+                std::transform(baseName.begin(), baseName.end(), std::back_inserter(lowerBaseName), ::tolower);
+
+                if (lowerBaseName == lowerModuleName) {
+                    return baseName;
+                }
+            }
+        }
+    }
+
+    return ext::nullopt;
+}
+
+static bool
+ResolveModuleMap(Phase::Environment const &phaseEnvironment, Phase::Context *phaseContext, pbxsetting::Environment const &environment)
```

Comment:
Ah! I only see one real choice: passing a copy separate from the caller (i.e. pass-by-value), or passing a pointer to change the caller's value (i.e. pass-by-reference). Then the question is what the best way is to implement each of those.

For passing a copy, a const reference has the same semantics of passing a copy — but without the performance cost of actually copying. So a const reference is just an optimization or implementation detail of the semantics of pass by value. For passing a pointer, both options have the same performance. So it falls back to the argument about which is most obvious to the caller about what might be mutated.

The other two cases, const pointer and non-const reference, end up not fitting into either group, so I don't use them much. Unused features are what make C++ fun, I guess.